client/checkout/utils/request.js

```diff
@@ -1,9 +1,21 @@
-export default async function request( url, args, options ) {
-	const data = new FormData();
-	for ( const key in args ) {
-		data.append( key, args[ key ] );
```

Comment:
I had to update this function to consider the nested objects too when translating an object to formData. I discovered that this function was not considering the nested fields when [this](https://github.com/Automattic/woocommerce-payments/blob/061efdc2bc6cfdd7eff7542f045364631cbab725/client/checkout/platform-checkout/express-button/use-express-checkout-product-handler.js#L70) `paymentRequestAddToCart` function was failing to add the product to the cart because the [attribute](https://github.com/Automattic/woocommerce-payments/blob/061efdc2bc6cfdd7eff7542f045364631cbab725/client/checkout/platform-checkout/express-button/use-express-checkout-product-handler.js#L44) field was not a proper form field (it was coming as `[object object]`.